types/applications/dialog.d.ts

```diff
@@ -0,0 +1,328 @@
+/**
+ * Create a modal dialog window displaying a title, a message, and a set of buttons which trigger callback functions.
+ *
+ * @example <caption>Constructing a custom dialog instance</caption>
+ * ```typescript
+ * let d = new Dialog({
+ *  title: "Test Dialog",
+ *  content: "<p>You must choose either Option 1, or Option 2</p>",
+ *  buttons: {
+ *   one: {
+ *    icon: '<i class="fas fa-check"></i>',
+ *    label: "Option One",
+ *    callback: () => console.log("Chose One")
+ *   },
+ *   two: {
+ *    icon: '<i class="fas fa-times"></i>',
+ *    label: "Option Two",
+ *    callback: () => console.log("Chose Two")
+ *   }
+ *  },
+ *  default: "two",
+ *  render: html => console.log("Register interactivity in the rendered dialog"),
+ *  close: html => console.log("This always is logged no matter which option is chosen")
+ * });
+ * d.render(true);
+ * ```
+ */
+declare class Dialog extends Application {
+  /**
+   * @param data    - An object of dialog data which configures how the modal window is rendered
```

Comment:
What about the `data` property? Should we include it in the type definition?

Reply:
you are absolutely right, I completely overlooked that.